app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Hashable, Any
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.length = 0
+        self.hash_table: list[None | list[Node]] = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length / self.capacity >= self.load_factor:
```

Comment:
The condition for resizing the hash table should use '>' instead of '>=' to ensure that resizing occurs only when the load factor is exceeded, not when it is exactly equal.